crates/aptos-rosetta/src/block.rs

```diff
@@ -44,42 +43,78 @@ async fn block(request: BlockRequest, server_context: RosettaContext) -> ApiResu
     let rest_client = &server_context.rest_client;
 
     // Retrieve by block or by hash, both or neither is not allowed
-    let response = match (
+    let (parent_transaction, transaction): (Transaction, _) = match (
         &request.block_identifier.index,
         &request.block_identifier.hash,
     ) {
-        (Some(version), None) => rest_client.get_transaction_by_version(*version).await?,
+        (Some(version), None) => {
```

Comment:
Does Rosetta need to fetch blocks all the way back to the Genesis block?

Reply:
Yes, this is step one.  I will need to add the API to tell how far back is pruned, otherwise it assumes you can go back to genesis.  We have to provide genesis at later points, so I need to make sure we have a way to get that.